core/src/main/java/org/infinispan/marshall/protostream/impl/MarshallableUserObject.java

```diff
@@ -0,0 +1,144 @@
+package org.infinispan.marshall.protostream.impl;
+
+import java.io.IOException;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.marshall.MarshallingException;
+import org.infinispan.commons.marshall.ProtoStreamTypeIds;
+import org.infinispan.commons.util.Util;
+import org.infinispan.protostream.ImmutableSerializationContext;
+import org.infinispan.protostream.RawProtoStreamReader;
+import org.infinispan.protostream.RawProtoStreamWriter;
+import org.infinispan.protostream.RawProtobufMarshaller;
+import org.infinispan.protostream.annotations.ProtoFactory;
+import org.infinispan.protostream.annotations.ProtoField;
+import org.infinispan.protostream.annotations.ProtoTypeId;
+import org.infinispan.protostream.impl.WireFormat;
+
+/**
+ * A wrapper message used by ProtoStream Marshallers to allow user objects to be marshalled/unmarshalled by the configured
+ * user marshaller via a ProtoStream Marshaller instance.
```

Comment:
We should probably document why some of these things are noops or never invoked. I don't know anything about how this class works with the the underlying protostream.

Reply:
I am assuming it is used only by the `RawProtobufMarshaller` but why are the annotations needed then?

Reply:
Yeah more documentation was definitely required for this class :slightly_smiling_face:. I've tried to explain as much as possible in the Javadocs.